### ISSUE TITLE
Error in logcat when searching update with latest beta version (v12.53)

### ISSUE THREAD

**Issue by afdev82** [yuliskov/SmartTubeNext#893]
I just updated the app to the v12.53 and I see this error in logcat if I try to manually search for an update:

```
09-14 16:30:55.809 1437-1809/com.liskovsoft.smarttubetv.beta E/OkHttpTLSCompat: Error while setting TLS 1.2 Unable to extract the trust manager on AndroidPlatform, sslSocketFactory is class com.liskovsoft.sharedutils.okhttp.Tls12SocketFactory
09-14 16:31:01.503 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err: javax.net.ssl.SSLHandshakeException: Handshake failed
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:390)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:351)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:310)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:178)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:236)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:109)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:77)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.Transmitter.newExchange$okhttp(Transmitter.kt:162)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:35)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
09-14 16:31:01.505 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:82)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at okhttp3.RealCall.execute(RealCall.kt:66)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.sharedutils.okhttp.OkHttpManager.doOkHttpRequest(OkHttpManager.java:167)
09-14 16:31:01.506 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.sharedutils.okhttp.OkHttpManager.doOkHttpRequest(OkHttpManager.java:112)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.sharedutils.okhttp.OkHttpHelpers.doOkHttpRequest(OkHttpHelpers.java:42)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.appupdatechecker2.other.downloadmanager.DownloadManager.doDownload(DownloadManager.java:82)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.appupdatechecker2.other.downloadmanager.DownloadManager.enqueue(DownloadManager.java:247)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.appupdatechecker2.core.AppVersionChecker$GetVersionJsonTask.getJSON(AppVersionChecker.java:251)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.appupdatechecker2.core.AppVersionChecker$GetVersionJsonTask.doInBackground(AppVersionChecker.java:238)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.liskovsoft.appupdatechecker2.core.AppVersionChecker$GetVersionJsonTask.doInBackground(AppVersionChecker.java:220)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at android.os.AsyncTask$2.call(AsyncTask.java:292)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
09-14 16:31:01.507 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at java.lang.Thread.run(Thread.java:818)
09-14 16:31:01.508 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err: Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x9c460600: Failure in SSL library, usually a protocol error
09-14 16:31:01.508 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:770 0xaf862fbd:0x00000000)
09-14 16:31:01.509 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
09-14 16:31:01.509 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err:     at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:318)
09-14 16:31:01.509 1437-1809/com.liskovsoft.smarttubetv.beta W/System.err: 	... 34 more
09-14 16:31:01.509 1437-1809/com.liskovsoft.smarttubetv.beta E/OkHttpManager: Handshake failed
09-14 16:31:01.509 1437-1809/com.liskovsoft.smarttubetv.beta E/AppVersionChecker: Error: bad response Error: bad response
09-14 16:31:01.550 1437-1809/com.liskovsoft.smarttubetv.beta E/OkHttpTLSCompat: Error while setting TLS 1.2 Unable to extract the trust manager on AndroidPlatform, sslSocketFactory is class com.liskovsoft.sharedutils.okhttp.Tls12SocketFactory
09-14 16:31:02.181 1437-1809/com.liskovsoft.smarttubetv.beta E/AppVersionChecker: Value <!DOCTYPE of type java.lang.String cannot be converted to JSONObject Value <!DOCTYPE of type java.lang.String cannot be converted to JSONObject
```

After some research I found [this issue](https://github.com/square/okhttp/issues/2323) in okhttp, but I don't know if it's the same or related.
Could you confirm it's something related with the app itself and not with my platform? I'm using Android v5.1.1.
The update was working in v12.51.

Many thanks for your support.

**Comment by WolfganP** [comment]
From https://github.com/yuliskov/SmartTubeNext/blob/master/common/src/main/res/values/update_urls.xml it seems the update json file at https://github.com/yuliskov/SmartYouTubeTV/releases/download/beta/smarttube_beta.json has some issues.

**Comment by afdev82** [comment]
I think the latest beta should use the other url at: https://github.com/yuliskov/SmartTubeNext/blob/master/common/src/stbeta/res/values/update_urls.xml as discussed in Telegram

**Comment by afdev82** [comment]
Fixed by v12.54 Beta: https://github.com/yuliskov/SmartTubeNext/commit/78c983e33123e913790af518ef33e8ba87245e0f
